src/app/core/header/header.component.html

```diff
@@ -5,10 +5,10 @@
 
   <ul class="nav justify-content-end">
     <li class="nav-item">
-      <p class="nav-link">{{getUserInfo()}}</p>
+      <p class="nav-link">{{userFirstName}} {{userLastName}}</p>
     </li>
     <li class="nav-item"
-        [ngSwitch]="isAuthenticated()">
+        [ngSwitch]="isAuthenticated">
```

Comment:
Sometimes it may be helpful to use `async` pipe to avoid redundant subscription in your .ts file